iceoryx_hoofs/include/iceoryx_hoofs/cxx/unique_ptr.hpp

```diff
@@ -93,26 +86,47 @@ class unique_ptr
     /// @brief release Release ownership of the underlying pointer.
     /// @return Pointer to the managed object or nullptr if none owned.
     ///
-    T* release() noexcept;
+    static T* release(unique_ptr&& ptrToBeReleased) noexcept;
 
     ///
     /// @brief reset Reset the unique pointer to take ownership of the given pointer.
     /// @details Any previously owned objects will be deleted. If no pointer given then points to nullptr.
```

Comment:
With the current changes one have to provide a pointer and this must not be a nullptr.